SSA/Core/EDSLNested.lean

```diff
@@ -0,0 +1,257 @@
+import SSA.Core.WellTypedFramework
+/-
+an EDSL that elaborates expressions with nesting
+allowed into SSA.
+
+Authors: Siddharth Bhat
+-/
+namespace EDSL2
+open Lean Elab Macro
+
+declare_syntax_cat dsl_op2
+scoped syntax "[dsl_op2|" dsl_op2 "]" : term
+syntax dsl_var2 := "%v" num
+declare_syntax_cat dsl_expr2
+declare_syntax_cat dsl_region2
+declare_syntax_cat dsl_stmt2
+
+/-
+This is a fancier syntax, which allows expressions in the
+intermediate nodes. This is elaborated into SSA, with temporary
+variable names.
+-/
+syntax dsl_var2 : dsl_expr2
+scoped syntax "()" : dsl_expr2
+scoped syntax "(" dsl_expr2 ")" : dsl_expr2
+scoped syntax "(" dsl_expr2 "," dsl_expr2 ")" : dsl_expr2
+scoped syntax "(" dsl_expr2 "," dsl_expr2 "," dsl_expr2 ")" : dsl_expr2
```

Comment:
@alexkeizer Our expression language only alows for tuples and triples, so if you want to pass four arguments you have to either make it a triple with a tuple somewhere inside, or a tuple of tuples (4 different options for the same 😬), etc..

Reply:
Is this a design choice on our part, our determined by some external factor (MLIR design or so)? It feels a bit arbitrary to support pairs and triples, but not arbitrary n-ary tuples.

Mathlib upstreaming is a bit slow, so I wouldn't mind trying to see if I can generalize this expression language, if you think it could be valuable to have

Reply:
Maybe let's discuss this over zulip?